terraform-plans/modules/GitHub/settings/main.tf

```diff
@@ -0,0 +1,87 @@
+terraform {
+  required_providers {
+    github = {
+      source  = "integrations/github"
+      version = "~> 6.0"
+    }
+  }
+}
+
+provider "github" {
+  owner = var.owner
+  app_auth {} # using environmet variables for auth
+}
+
+resource "github_repository" "repo" {
+  name = var.repository
+
+  has_issues      = true
+  has_projects    = false
+  has_wiki        = false
+  has_discussions = false
+
+  allow_merge_commit = false
+  allow_squash_merge = true
+  allow_rebase_merge = false
+  allow_auto_merge   = false
+
+  allow_update_branch    = true
+  delete_branch_on_merge = true
+
+}
+
+data "github_team" "admins" {
+  slug = "soleng-admin"
+}
+
+data "github_team" "reviewers" {
+  slug = "soleng-reviewers"
+}
+
+data "github_team" "engineering" {
+  slug = "solutions-engineering"
+}
+
+resource "github_repository_collaborators" "repo_collaborators" {
+  repository = var.repository
+
+  team {
+    permission = "admin"
+    team_id    = data.github_team.admins.id
+  }
+
+  team {
+    permission = "maintain"
+    team_id    = data.github_team.reviewers.id
+  }
+
+  team {
+    permission = "push"
+    team_id    = data.github_team.engineering.id
+  }
+}
+
+resource "github_branch_protection" "branch_protection" {
+  repository_id                   = github_repository.repo.node_id
+  pattern                         = var.branch
+  enforce_admins                  = true
```

Comment:
What is enforce_admin?

Reply:
Can be found here: https://registry.terraform.io/providers/integrations/github/latest/docs/resources/branch_protection

> (Optional) Boolean, setting this to true enforces status checks for repository administrators.

Reply:
To be honest, I have no idea what it's actually doing, but I saw that in all examples.